### PR TITLE
DAOS-12106 Improve branch prediction with likely and unlikely

### PR DESCRIPTION
**Pull request by zjf2012**
The events allocation is also enhanced to reduce number of allocations.

Signed-off-by: jiafu zhang <jiafu.zhang@intel.com>

### CODE REVIEW

**Review by johannlombardi — DISMISSED** [review]

**Comment by zjf2012** [comment]
@mchaarawi please help review

File: src/client/java/daos-java/src/main/native/io_daos_DaosClient.c [hunk 1]
```diff
@@ -363,44 +363,29 @@ Java_io_daos_DaosClient_createEventQueue(JNIEnv *env,
...
-
-			asprintf(&msg, "Failed to allocate %d th event.",
-				 i);
-			rc = 1;
-			throw_base(env, msg, rc, 1, 0);
-			goto fail;
-		}
-		rc = daos_event_init(&eq->events[i]->event, eqhdl, NULL);
-		if (rc) {
+	for (; i < nbrOfEvents; i++) {
```
Comment (mchaarawi) [inline src/client/java/daos-java/src/main/native/io_daos_DaosClient.c:366]: minor, but style wise it's always better to have the initialization of the counter here.

**Review by mchaarawi — APPROVED** [review]

File: src/client/java/daos-java/src/main/native/io_daos_DaosClient.c (hunk 1 above)
Comment (zjf2012) [reply src/client/java/daos-java/src/main/native/io_daos_DaosClient.c:366]: thanks

**Comment by zjf2012** [comment]
@johannlombardi please help merge.

**Comment by zjf2012** [comment]
@johannlombardi could you please help merge it?

**Review by mchaarawi — APPROVED** [review]

**Pull request merged**